web/packages/teleterm/src/services/tshd/createClient.ts

```diff
@@ -903,105 +955,108 @@ export function createTshdClient(
       });
     },
     getUserPreferences(
-      params: api.GetUserPreferencesRequest.AsObject,
+      params: api.GetUserPreferencesRequest,
       abortSignal?: types.TshAbortSignal
-    ): Promise<api.UserPreferences.AsObject> {
+    ): Promise<api.UserPreferences> {
       return withAbort(abortSignal, callRef => {
-        const req = new api.GetUserPreferencesRequest().setClusterUri(
-          params.clusterUri
-        );
+        const req = api.GetUserPreferencesRequest.create({
+          clusterUri: params.clusterUri,
+        });
 
         return new Promise((resolve, reject) => {
           callRef.current = tshd.getUserPreferences(req, (err, response) => {
             if (err) {
               reject(err);
             } else {
-              const res = response.toObject();
-              resolve(res.userPreferences);
+              resolve(response!.userPreferences!);
             }
           });
         });
       });
     },
     updateUserPreferences(
-      params: api.UpdateUserPreferencesRequest.AsObject,
+      params: api.UpdateUserPreferencesRequest,
       abortSignal?: types.TshAbortSignal
-    ): Promise<api.UserPreferences.AsObject> {
-      const userPreferences = new UserPreferences();
-      if (params.userPreferences.clusterPreferences) {
-        userPreferences.setClusterPreferences(
-          new ClusterUserPreferences().setPinnedResources(
-            new PinnedResourcesUserPreferences().setResourceIdsList(
-              params.userPreferences.clusterPreferences.pinnedResources
-                .resourceIdsList
-            )
-          )
-        );
-      }
+    ): Promise<api.UserPreferences> {
+      const userPreferences = UserPreferences.create();
+      if (params.userPreferences) {
```

Comment:
That seems like a leftover from strict null checks. This check here probably doesn't do anything bad, but I guess I'd remove it just to keep the implementation as close to the previous one as possible?

Reply:
> I'd remove it just to keep the implementation as close to the previous one as possible?

Agreed, undone